CMakeLists.txt

```diff
@@ -868,6 +868,13 @@ endforeach()
 # Convenience variable for linkers
 get_property(ZEPHYR_LINK_LIBS_PROPERTY TARGET zephyr_interface PROPERTY LINK_LIBS)
 
+# Remove any non-file options which aren't libraries from this property
+foreach(zephyr_link_lib ${ZEPHYR_LINK_LIBS_PROPERTY})
+ if(zephyr_link_lib MATCHES "^-.*" AND NOT zephyr_link_lib MATCHES "^-l.*")
```

Comment:
and this already gives a hint why this is a dangerous path to be taken.

I believe the general CMake lib linking handling can take care of this for us, but I would need to know the exact issues faced / how I can reproduce, then I can provide a PR proposal adjustment, I believe that will be the most efficient way to tackle this.
Although I'm also busy with HWMv2, then I believe this is worth (and most efficient) to be spending time on.
Also feel free to ping on Discord.

Reply:
> and this already gives a hint why this is a dangerous path to be taken.

No, this should be replaced by separating this into two functions, one for libraries and the other for link options. I didn't want to get started on that as it's more invasive than this kludge.

Reply:
I guess the real question is whether this is worth the effort at all. Do we want to get away from `--whole-archive` as the way to resolve the inherent circular dependencies when linking Zephyr? Or do we just want to leave things as they are and expect application developers to configure the system to explicitly exclude bits they don't use?